pageserver/src/layered_repository.rs

```diff
@@ -281,12 +281,21 @@ impl Repository for LayeredRepository {
         // concurrently removes data that is needed by the new timeline.
         let _gc_cs = self.gc_cs.lock().unwrap();
 
+        // Besides GC lock, branch creation task doesn't need to hold the `layer_removal_cs` lock,
+        // hence doesn't need to wait for compaction/GC because it ensures that the starting LSN
+        // of the child branch is not out of scope in the middle of the creation task by
+        // 1. holding the GC lock to prevent overwritting timeline GC data
+        // 2. checking both the latest GC cutoff LSN and latest GC info of the source timeline
+        // to avoid initializing the new branch using data removed by past GC iterations
+        // or in-queue GC iterations.
+
```

Comment:
It's a bit weird to explain in great detail why this *doesn't* need to hold `layer_removal_cs` lock. Also, this seems like a run-on sentence.